Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2024 Colin Jones. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colin Jones
+-/
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Tactic.NormNum.Prime
+import Mathlib.Algebra.GeomSum
+import Mathlib.Tactic.FinCases
+
+/-!
+## Factorisation properties of natural numbers
+This file defines abundant, pseudoperfect, deficient, and weird numbers and formalizes their
+  relations with prime and perfect numbers.
+
+## Main Definitions
+* `Nat.Abundant`: a natural number `n` is _abundant_ if the sum of its proper divisors are greater
+  than `n`
+* `Nat.Pseudoperfect`: a natural number `n` is _pseudoperfect_ if a subset of its proper divisors
+  equals `n`
+* `Nat.Deficient`: a natural number `n` is _deficient_ if the sum of its proper divisors is less
+  than `n`
+* `Nat.WeirdNumber`: a natural number is _weird_ if it is both abundant and not pseudoperfect
+
+## Main Results
+* `Nat.deficient_or_perfect_or_abundant`: a positive natural number is either deficient,
+  perfect, or abundant
+* `Nat.prime_deficient`: all prime natural numbers are deficient
+* `Nat.exists_infinite_deficients`: there are infinitely many deficient numbers
+* `Nat.prime_pow_deficient`: any natural number power of a prime is deficient
+
+## Implementation Details
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+
+abundant, deficient, weird, pseudoperfect
+-/
+
+open BigOperators Finset
+
+namespace Nat
+
+variable {n m : ℕ}
+
+/-- `n : ℕ` is _abundant_ if and only if the sum of the proper divisors of `n` is greater than `n`
+  -/
+def Abundant (n : ℕ) : Prop := ∑ i in properDivisors n, i > n
+
+/-- `n : ℕ` is _deficient_ if and only if the sum of the proper divisors of `n` is less than `n` -/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- `n : ℕ` is _pseudoperfect_ if and only if there exists a subset of the proper divisors of n
+  such that the sum of that subset is equal to `n` and `n` is positive -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s : Finset ℕ, s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n
```

Comment:
```suggestion
  0 < n ∧ ∃ s ⊆ properDivisors n, ∑ i ∈ s, i = n
```
This breaks something later on but it's the idiomatic way to say the definition.